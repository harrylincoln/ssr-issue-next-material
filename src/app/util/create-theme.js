import { createMuiTheme } from '@material-ui/core/styles';

const breakpointValues = {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200,
};

export function themeCreator(primary = {}) {
    return createMuiTheme({
        breakpoints: { values: breakpointValues },
        typography: {
            useNextVariants: true,
        },
        pallete: {
            ...primary,
        },
    });
}
