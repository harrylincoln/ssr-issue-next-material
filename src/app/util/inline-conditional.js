export default (condition, truthy, falsy = null) => {
    if (condition) return [truthy];
    if (falsy === null) return [];
    return [falsy];
};
