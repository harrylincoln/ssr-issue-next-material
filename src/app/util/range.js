export default (length, offset = 0) =>
    Array(length)
        .fill(undefined)
        .map((_, i) => offset + i);
