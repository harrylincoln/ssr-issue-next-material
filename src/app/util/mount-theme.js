import { JssProvider, SheetsRegistry, ThemeProvider } from 'react-jss';
import { mount, shallow } from 'enzyme';

import { Provider } from 'react-redux';
import React from 'react';
import configureMockStore from 'redux-mock-store';
// theming
import primary from '../styles/themes/primary';
import { themeCreator } from './create-theme';
import thunk from 'redux-thunk';
import { createGenerateClassName } from '@material-ui/core/styles';

const mockStore = configureMockStore([thunk]);
const primaryTheme = themeCreator({ primary });

/**
 * JSS helper mount
 * Wraps the passed in component in Themeprovider and captures the styles generated
 * @param {React component} Component
 * @returns {Array} First item in array is the stylesheet generated by jss, the second is the rendered component for further expects
 */
export const themeMount = (Component, store = mockStore({})) => {
    const sheets = new SheetsRegistry();
    const ClonedComponent = React.cloneElement(Component, {
        _renderid: 'rendered-component' /* eslint-disable-line */,
    });

    const fullTree = mount(
        <JssProvider
            generateClassName={createGenerateClassName()}
            registry={sheets}
        >
            <ThemeProvider theme={primaryTheme}>
                <Provider store={store}>{ClonedComponent}</Provider>
            </ThemeProvider>
        </JssProvider>
    );

    const mountWithTheme = (tree, theme) => {
        const context = shallow(<ThemeProvider theme={theme} />, {
            lifecycleExperimental: true,
        })
            .instance()
            .getChildContext();

        return mount(tree, {
            context,
            childContextTypes: ThemeProvider.childContextTypes, // Needed so child components receive the theme prop
        });
    };

    const fullMountWithTheme = mountWithTheme(
        <Provider store={store}>
            <JssProvider
                generateClassName={createGenerateClassName()}
                registry={sheets}
            >
                {Component}
            </JssProvider>
        </Provider>,
        primaryTheme
    );

    return {
        styleSheets: sheets.toString(),
        component: fullTree.findWhere(
            (node) =>
                node.props()._renderid === /* eslint-disable-line */
                'rendered-component'
        ),
        fullRendered: fullTree,
        fullMountWithTheme,
        mockStore: store,
    };
};
