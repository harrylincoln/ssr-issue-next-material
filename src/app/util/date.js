import range from './range';

export const getMidnight = (date = new Date()) => {
    const clone = new Date(date);
    clone.setHours(0);
    clone.setMinutes(0);
    clone.setSeconds(0);
    clone.setMilliseconds(0);

    return clone;
};

export const getNextSaturday = (date) => {
    const currentDay = date.getDay();

    const daysUntilSaturday = currentDay === 6 ? 6 - currentDay : 7;

    const saturday = new Date(date);

    saturday.setDate(date.getDate() + daysUntilSaturday);

    return saturday;
};

export const addWeek = (date, weeks = 1) => {
    const clone = new Date(date);
    clone.setDate(date.getDate() + 7 * weeks);
    return clone;
};

export const getWeeks = (startDate, amount) =>
    range(amount, 1).map((week) => addWeek(startDate, week));

export const months = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
];

export const getOrdinal = (day) => {
    const ordinals = ['', 'st', 'nd', 'rd'];
    const remainder = day % 100;
    return (
        day +
        (remainder > 10 && remainder < 14
            ? 'th'
            : ordinals[remainder % 10] || 'th')
    );
};
