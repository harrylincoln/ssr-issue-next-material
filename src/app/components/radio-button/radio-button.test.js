import RadioButton from './radio-button';
import React from 'react';

import { themeMount } from '../../util/mount-theme';

describe('RadioButton', () => {
    it('should render correctly', () => {
        const props = {
            text: 'test title',
        };

        const { styleSheets, component } = themeMount(
            <RadioButton {...props}>Test</RadioButton>
        );

        expect(styleSheets.toString()).toMatchSnapshot();
        expect(component).toMatchSnapshot();
    });

    it('should be highlighted and description visible if select prop is true', () => {
        const props = {
            text: 'test title',
            selected: true,
        };
        const { component } = themeMount(
            <RadioButton {...props}>Test</RadioButton>
        );
        expect(component.find(RadioButton).instance().props.selected).toBe(
            true
        );
    });
});
