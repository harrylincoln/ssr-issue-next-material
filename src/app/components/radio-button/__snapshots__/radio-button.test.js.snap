// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should render correctly 1`] = `
".btnRadio-test {
  width: 100%;
  color: #383E42;
  cursor: pointer;
  border: 1px solid rgba(62,81,181,0);
  display: block;
  padding: 1rem 15px;
  position: relative;
  box-sizing: border-box;
  background: #818ED4;
  margin-bottom: .75rem;
}
.btnRadio-test:focus {
  outline: none;
}
.btnRadio-test:hover {
  background: #6E7ECD;
}
.btnRadio_selected-test {
  fill: #3E51B5;
  color: #3E51B5;
  background: white;
}
.btnRadio_selected-test:hover {
  background: white;
}"
`;

exports[`RadioButton should render correctly 2`] = `
Array [
  <WithStyles(RadioButton)
    _renderid="rendered-component"
    text="test title"
  >
    <RadioButton
      _renderid="rendered-component"
      classes={
        Object {
          "btnRadio": "btnRadio-test",
          "btnRadio_selected": "btnRadio_selected-test",
        }
      }
      id=""
      onClick={[Function]}
      selected={false}
      text="test title"
    >
      <button
        className="btnRadio-test"
        id=""
        key=""
        onClick={[Function]}
        type="button"
      >
        test title
      </button>
    </RadioButton>
  </WithStyles(RadioButton)>,
  <RadioButton
    _renderid="rendered-component"
    classes={
      Object {
        "btnRadio": "btnRadio-test",
        "btnRadio_selected": "btnRadio_selected-test",
      }
    }
    id=""
    onClick={[Function]}
    selected={false}
    text="test title"
  >
    <button
      className="btnRadio-test"
      id=""
      key=""
      onClick={[Function]}
      type="button"
    >
      test title
    </button>
  </RadioButton>,
]
`;
